src/Golang/providers.go

```diff
@@ -0,0 +1,24 @@
+package filemanager
+
+type PathProvider interface {
+	Combine(paths ...string) (string, error)
+	GetObjectName(path string) (string, error)
+	GetObjectParentPath(path string) (string, error)
+}
+
+type DirectoryManager interface {
+	GetPathProvider() PathProvider
+	CreateDir(path string) (*DirectoryDetail, error)
+	GetDir(path string) (*DirectoryDetail, error)
+	DirExists(path string) (bool, error)
+	DeleteDir(path string) error
+}
+
+type FileManager interface {
+	GetPathProvider() PathProvider
+	GetDirectoryManager() DirectoryManager
+	CreateFile(path string) (*FileDetail, error)
```

Comment:
@AshkanAbd Can we please follow the structure of the c# interface?
because we have to make the same documentation for all of the functions.

for example, use CreateFileAsync instead of CreateFile.

It's easy to read for all developers in any language and we follow one structure in every library.

1. Same documentation
2. Same code metric
3. Readable code
4. Convertable code
5. Same roadmap